tests/test_CompareEclipseRestart.cpp

```diff
@@ -0,0 +1,200 @@
+    if (ecl_file2 == nullptr) {
+        ecl_file2 = ecl_file_open(unrstFile2, ECL_FILE_CLOSE_STREAM);
+    }
+
+    if (ecl_file1 == nullptr) {
+        OPM_THROW(std::runtime_error, "Error opening first .UNRST-file.");
+        return false;
+    } 
+    else if (ecl_file2 == nullptr) {
+        OPM_THROW(std::runtime_error, "Error opening second .UNRST-file.");
+        return false;
+    }
+    return true;
+}
+
+
+
+void ReadUNRST::close() {
+    ecl_file_close(ecl_file1);
+    ecl_file_close(ecl_file2);
+}
+
+
+
+bool ReadUNRST::results(const char* keyword, std::vector<double>& absDeviation, std::vector<double>& relDeviation) const {
+    if (!ecl_file_has_kw(ecl_file1, keyword) || !ecl_file_has_kw(ecl_file2, keyword)) {
+        OPM_THROW(std::runtime_error, "The file does not have this keyword.");
+        return false;
+    }
+    const unsigned int occurrences1 = ecl_file_get_num_named_kw(ecl_file1, keyword);
+    const unsigned int occurrences2 = ecl_file_get_num_named_kw(ecl_file2, keyword);
+    if (occurrences1 != occurrences2) {
+        OPM_THROW(std::runtime_error, "Number of keyword occurrences are not equal.");
+        return false;
+    }
+    for (unsigned int index = 0; index < occurrences1; ++index) {
+        ecl_kw_type* ecl_kw1 = ecl_file_iget_named_kw(ecl_file1, keyword, index);
+        const unsigned int numActiveCells1 = ecl_kw_get_size(ecl_kw1);
+        ecl_kw_type* ecl_kw2 = ecl_file_iget_named_kw(ecl_file2, keyword, index);
+        const unsigned int numActiveCells2 = ecl_kw_get_size(ecl_kw2);
+        if (numActiveCells1 != numActiveCells2) {
+            OPM_THROW(std::runtime_error, "Number of active cells are different.");
+            return false;
+        }
+        std::vector<double> values1, values2; // Elements in the vector corresponds to active cells
+        values1.resize(numActiveCells1);
+        values2.resize(numActiveCells2);
+        ecl_kw_get_data_as_double(ecl_kw1, values1.data());
+        ecl_kw_get_data_as_double(ecl_kw2, values2.data());
+
+        for (unsigned int i = 0; i < values1.size(); i++) {
+            Deviation dev = calculateDeviations(values1[i], values2[i]);
+            if (dev.abs != -1) {
+                absDeviation.push_back(dev.abs);
+            }
+            if (dev.rel != -1) {
+                relDeviation.push_back(dev.rel);
+            }
+        }
+    }
+    return true;
+}
+
+
+
+Deviation ReadUNRST::calculateDeviations(double val1, double val2) const {
+    Deviation deviation;
+    if (val1 < 0) {
+        val1 = 0;
+        //std::cout << "Value 1 has a negative quantity." << std::endl; 
+    }
+    if (val2 < 0) {
+        val2 = 0;
+        //std::cout << "Value 2 has a negative quantity." << std::endl; 
+    }
+
+    if (!(val1 == 0 && val2 == 0)) {
+        deviation.abs = std::max(val1, val2) - std::min(val1, val2);
+        if (val1 != 0 && val2 != 0) {
+            deviation.rel = deviation.abs/static_cast<double>(std::max(val1, val2));
```

Comment:
Your right, no point in casting here :+1: The point of the if statement on line 170 is to avoid adding a relative deviation of 1 when one of the values are 0. Then the -1 initialization of the Deviation struct ensures that the deviationvectors are only updated when the Deviation doubles are different from -1. I guess there might be a better way of doing this though